### DISCUSSION TITLE
Javadoc definition appearing while I am typing the object name, but not when I hover

### DISCUSSION THREAD

**Discussion by pepper-lees** [scalameta/nvim-metals#587] [Q&A] [answered]
There's an object called `InModuleBody` in one of the libraries I'm using with this as its definition:
```scala
/** Used to inject code snippets to be evaluated in [[LazyModuleImp.instantiate]] in the current [[LazyModule.scope]].
  *
  * It can be used to create additional hardware outside of the [[LazyModule.children]],
  * connections other than the internal [[BaseNode]] connections,
  * or additional IOs aside from the [[AutoBundle]]
  */
object InModuleBody {
  def apply[T](body: => T): ModuleValue[T] = {
    // <code body>
  }
}
```
When I start typing it out, the Javadoc does appear for the object:
![image](https://github.com/scalameta/nvim-metals/assets/11487870/9f6fc5a0-12ce-4a45-9d20-0e7a79b1c171)

But when I use the object in code, and use the "hover" command, it seems to only show the `apply` signature and ignores all that helpful Javadoc at the top of the object's definition.

![image](https://github.com/scalameta/nvim-metals/assets/11487870/4945f9a9-a7a8-497c-909d-cf8d91001d48)

Also, if I use the "Signature help" command, nothing appears. Why is that the case? Is there a command I can use that will show the Javadoc as expected?

**Comment by ckipp01** [comment]
Hey @JL102 so I think the behavior here about showing the doc for `apply` is actually expected. When you do a hover on `InModuleBody` here this is actually the apply method that is getting called, so I'd sort of expect it to show you exactly what it's showing you. @tgodzik this is correct right?

As for the "Signature help" command, it depends on where you're triggering it. I just tested this and if you're triggering "signature help" on `InModuleBody` it won't give you anything. _However_ if you trigger it **in** the actual parameter it will. Meaning if you would move down one line and trigger it where you have `// <code>` you should see the signature help appear.

**Reply by tgodzik** [reply, marked as answer]
Actually, that looks like a bug. Doing `InModuleBody.apply(123)` and hovering on `InModuleBody` also doesn't show anything

Reported https://github.com/scalameta/metals/issues/5431

**Reply by pepper-lees** [reply]
Thanks! 😃
